spec.emu

```diff
@@ -52,13 +54,17 @@ contributors:
         </h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It returns a string representing a |Pattern| for matching _c_. If _c_ is white space or an ASCII punctuator, the returned value is an escape sequence (corresponding with |HexEscapeSequence| if possible, or otherwise with |RegExpUnicodeEscapeSequence|). Otherwise, the returned value is a string representation of _c_ itself.</dd>
+          <dd>It returns a string representing a |Pattern| for matching _c_. If _c_ is white space or an ASCII punctuator, the returned value is an escape sequence. Otherwise, the returned value is a string representation of _c_ itself.</dd>
         </dl>
 
         <emu-alg>
-          1. Let _punctuators_ be the string-concatenation of *"(){}[]|,.?\*+-^$=<>/#&!%:;@~'`"*, the code unit 0x0022 (QUOTATION MARK), and the code unit 0x005C (REVERSE SOLIDUS).
-          1. Let _toEscape_ be StringToCodePoints(_punctuators_).
-          1. If _toEscape_ contains _c_ or _c_ is matched by |WhiteSpace|, then
+          1. If _c_ is matched by |SyntaxCharacter| or _c_ is U+002F (SOLIDUS), then
+            1. Return the string-concatenation of 0x005C (REVERSE SOLIDUS) and UTF16EncodeCodePoint(_c_).
+          1. Else if _c_ is the code point listed in some cell of the "Code Point" column of <emu-xref href="#table-controlescape-code-point-values"></emu-xref>, then
+            1. Return the string-concatenation of 0x005C (REVERSE SOLIDUS) and the string in the "ControlEscape" column of the row whose "Code Point" column contains _c_.
+          1. Let _otherPunctuators_ be the string-concatenation of *",-=<>#&!%:;@~'`"* and the code unit 0x0022 (QUOTATION MARK).
+          1. Let _toEscape_ be StringToCodePoints(_otherPunctuators_).
+          1. If _toEscape_ contains _c_, _c_ is matched by |WhiteSpace| or |LineTerminator|, or _c_ is in the inclusive interval from U+D800 to U+DFFF (i.e. _c_ is a leading surrogate or trailing surrogate), then
```

Comment:
cf. [QuoteJSONString](https://tc39.es/ecma262/multipage/structured-data.html#sec-quotejsonstring)
```suggestion
          1. If _toEscape_ contains _c_, _c_ is matched by |WhiteSpace| or |LineTerminator|, or _c_ has the same numeric value as a leading surrogate or trailing surrogate, then
```

Reply:
can't we *just* say `_c_ is a leading surrogate or trailing surrogate`, since that's dfn'd, without talking about numeric values?

Reply:
No, because "leading surrogate" and "trailing surrogate" apply to code units rather than code points and we don't want to conflate the two.